src/dataToEsm.js

```diff
@@ -44,6 +44,9 @@ export default function dataToNamedExports (data, options = {}) {
 	const _ = options.compact ? '' : ' ';
 	const n = options.compact ? '' : '\n';
 	const declarationType = options.preferConst ? 'const' : 'var';
+	
+	if (options.namedExports === false || typeof data !== 'object' || Array.isArray(data))
```

Comment:
If we actually allow for non-object types to be passed here, we should add a check for `null` as well.